pkg/cloudprovider/aws/amifamily/ami.go

```diff
@@ -57,11 +65,16 @@ func (p *AMIProvider) Get(ctx context.Context, provider *awsv1alpha1.AWS, nodeRe
 	if err != nil {
 		return nil, err
 	}
+	// Keep track of usedInstancePools so we don't add duplicates to the EC2 API Call
+	usedInstancePools := sets.NewString()
 	if len(amiRequirements) > 0 {
+		// Iterate through AMIs in order of creation date to get use latest AMI
+		amis := sortAMIsByCreationDate(amiRequirements)
 		for _, instanceType := range nodeRequest.InstanceTypeOptions {
-			for amiID, requirements := range amiRequirements {
-				if err := instanceType.Requirements().Compatible(requirements); err == nil {
-					amiIDs[amiID] = append(amiIDs[amiID], instanceType)
+			for _, ami := range amis {
+				if err := instanceType.Requirements().Compatible(amiRequirements[ami]); err == nil && !usedInstancePools.Has(instanceType.Name()) {
+					amiIDs[ami.AmiID] = append(amiIDs[ami.AmiID], instanceType)
+					usedInstancePools.Insert(instanceType.Name())
```

Comment:
Could you consider just `break` here instead of keeping track of the usedInstancePools? The instanceType range should imply that we won't be repeating instance types?

Reply:
Good point. I played around with some other iteration orders and didn't realize I could do this. Thanks!